### PR TITLE
simple deploy_marketplace.sh

### PR DESCRIPTION
**Pull request by AsherShoshan**
A simple deploy_Marketplace.sh script, which does not include the creation of operatorSource nor catalogSourceConfig.
In upstream/downstream openshift cluster creation of operatorSorce/catalogSource is not required (redhat-operators is used)
For QE/Dev environment, custom opsrc/catsrc are provided in an earlier step of cluster creation.

### CODE REVIEW

File: deploy/deploy_marketplace.sh [hunk 1]
```diff
@@ -2,195 +2,72 @@
 
 set -ex
 
-RED='\033[0;31m'
-NO_COLOR='\033[0m'
```
Comment (dankenigsberg) [inline deploy/deploy_marketplace.sh]: this change is irrelevant. If you think that this should be dropped, please do it in a separate PR.

File: deploy/deploy_marketplace.sh [hunk 2]
```diff
@@ -2,195 +2,72 @@
...
-
 globalNamespace=`oc -n openshift-operator-lifecycle-manager get deployments catalog-operator -o jsonpath='{.spec.template.spec.containers[].args[1]}'`
 echo "Global Namespace: ${globalNamespace}"
+GLOBAL_NAMESPACE="${GLOBAL_NAMESPACE:-$globalNamespace}"
```
Comment (dankenigsberg) [inline deploy/deploy_marketplace.sh]: I see you do not introduce this in this PR, but the previous line printed one value, and now you are going to use another. This would confuse the reader.

File: deploy/deploy_marketplace.sh (hunk 1 above)
Comment (AsherShoshan) [reply deploy/deploy_marketplace.sh]: Because when you have a cluster (clean one), you already have 3 catalogsources: redhat-operators, community, and certified-operators. So no need for preparation.
In case of QE/Dev, custom catalogsources are prepared earlier, and already used for other operators (not just CNV)

Should someone, needs a custom catsrc, then he can't use the "full" script as it's not supporting the creation of a plain catalogSource (ONLY by operatorSource, from QUAY).

IMO, an OLM catalogSource creation script, should be added to repo as an extra utility, and to support plain/opsrc catsrc creation.

Comment (dankenigsberg) [reply deploy/deploy_marketplace.sh]: I don't like long scripts myself. If you want to split out some logic to another script, please do it. But please take care of existing use cases

Comment (AsherShoshan) [reply deploy/deploy_marketplace.sh]: As mentioned before, for 99.99% of deploy_marketplace.sh, creation of catalogSOurce is not required.  For other cases, the split is in a new script (with old content) deploy_operatoSource.sh.

Comment (tiraboschi) [reply deploy/deploy_marketplace.sh]: Currently we already have deploy_imageregistry.sh (to consume the bundle image as a local registry) and deploy_marketplace.sh (to deploy pointing to a remote app-registry) and we want to unify them (to avoid having to duplicate each fix to both of them).
Introducing a third script is definitively not an option.

Comment (AsherShoshan) [reply deploy/deploy_marketplace.sh]: deploy_imageregistry is creating catalogSource not an operatorSource from QUAY.
Also this one should be splitted..
Sure, it's a good idea to unify them (just for catsrc/opsrc creation)

**Comment by tiraboschi** [comment]
We are going to use kustomize for this, see https://github.com/kubevirt/hyperconverged-cluster-operator/pull/463
